### PR TITLE
Add math expression parser for markdown

### LINKED ISSUE #14155: CodeMirror 6 Markdown parser does not highlight LaTeX syntax
## Description

LaTeX highlighting in Markdown cell is missing. A parser can be implemented as explained [here](https://codemirror.net/examples/lang-package/)

Extracted from https://github.com/jupyterlab/jupyterlab/issues/12812

### PR DESCRIPTION
**Pull request by fcollonval**
## References

Fixes #14155

## Code changes

Add custom Markdown extension to parse LaTeX mathematical expressions.

## User-facing changes

Mathematical expression are highlighted in CodeMirror Markdown editors.

Before
image

After
image

## Backwards-incompatible changes

None

### CODE REVIEW

File: packages/codemirror/src/codemirror-ipythongfm.ts [hunk 1]
```diff
@@ -1,109 +0,0 @@
-// Copyright (c) Jupyter Development Team.
```
Comment (bollwyvl) [inline packages/codemirror/src/codemirror-ipythongfm.ts:1]: so long and thanks for all the GFM!

File: packages/codemirror-extension/src/services.tsx [hunk 2]
```diff
@@ -60,10 +62,16 @@ export const languagePlugin: JupyterFrontEndPlugin<IEditorLanguageRegistry> = {
       name: 'ipythongfm',
       mime: 'text/x-ipythongfm',
       load: async () => {
-        // TODO: add support for LaTeX
-        const m = await import('@codemirror/lang-markdown');
+        const [m, tex] = await Promise.all([
+          import('@codemirror/lang-markdown'),
+          import('@codemirror/legacy-modes/mode/stex')
```
Comment (bollwyvl) [inline packages/codemirror-extension/src/services.tsx:67]: good find!

**Review by jtpio — APPROVED** [review]
Thanks!

**Comment by jtpio** [comment]
Looks like the UI tests check is also failing on other PRs and is likely not related.

**Pull request closed without merging**